src/05-derivatives/04-functional-derivatives.md

```diff
@@ -47,51 +47,62 @@ The following metadata JSON fields are valid for derivative maps:
 | Threshold      | String describing threshold used for determining graph edges.             | `dcb`, `dcw`, `ecb`, `ecw` |
 | Method         | String describing method used to calculate measure.                       | `dcb`, `dcw`, `ecb`, `ecw` |
 
-## Time series and regressors
+## Time series
 
-For the purposes of this section, a time series or regressor is a numeric series
-of values corresponding to volumes in a BOLD series. Time series is specifically
-used to indicate an aggregate time series within an ROI, while regressor is used
-as a more general term indicating series that may be derived from other sources,
-such as head motion estimates or physiological recordings.
+A time series is an ordered series of numeric values in chronological order.
+For the purposes of this section, a "regressor" is a time series with one entry
+per volume in a BOLD series.
 
-Regressors will generally be stored as tables, with a row of column headers
-indicating the name of the regressor. In the case where every voxel has a
-regressor, then the data should be stored in a 4D NIfTI file.
+Time series will generally be stored as tables, with a row of column headers
+indicating the name of the series.
+In the case where every voxel has a time series, then the data should be stored
+in a 4D NIfTI file.
+
+All time series files MUST be accompanied by a data dictionary in JSON format,
+containing the following sampling information:
+
+| Field name | Definition |
+| ---------- | ---------- |
+| SamplingFrequency | REQUIRED. Sampling frequency (in Hz) of all columns in the file. Special value `"TR"` indicates one sample per volume of a corresponding BOLD series. |
+| StartTime | OPTIONAL. Time in seconds of the first sample, relative to the run start. (Default: 0s) |
 
-Tabular regressor files MUST be accompanied by a data dictionary in JSON format,
+The `"TR"` sampling frequency serves to indicate that no resampling is needed
+to use the series as a regressor for BOLD data, including BOLD series with
+non-uniform sampling, such as clustered sparse acquisition.
+
+Tabular time series files MUST be accompanied by additional metadata,
```

Comment:
This is a bit confusing to me coming after the data dictionary section. Maybe move it up, and then note that a data dictionary must be included in the metadata, following the specifications laid out in the core BIDS spec?

Reply:
I can adjust this a little. I agree this part doesn't flow very well at the moment.

Reply:
Okay, I'm having trouble writing this in spec-ese, but want to lay out my thoughts:

1) SamplingFrequency is required for all files, whether TSV or NIfTI (should probably extend to GIFTI/CIFTI-2, but I'm thinking we should leave that for a later PR).
2) NIfTIs should be treated in this context as a single TSV column, in that all the metadata except the spatial position of each voxel is identical.
3) All of the usual TSV metadata applies to each column/NIfTI.

Just want to make sure this sounds reasonable to everyone... Assuming so, is it better to somehow rhetorically group TSVs and NIfTIs prior to discussing metadata, or discuss TSVs and then discuss NIfTIs? The TSV metadata is per-column (except the fields I'm adding), while the *same* metadata should apply to the NIfTI file directly. Which is a little awkward to write.

Upgrading basic TSV metadata from optional to recommended or required is a separate question, but do people have an opinion there?

Reply:
CIFTI-2 has sampling frequency in the file meta data. But it might make sense to make it optional in the json for consistency with other formats?

Reply:
@effigies LGTM. I'd go with discussing file format (TSV or nifti) first, and metadata second. (And it's probably worth explicitly saying in the latter section that all of the same metadata rules apply to both TSV and nifti.)

@edickie that would make for a really bad time implementation-wise... e.g., pybids ideally shouldn't have to understand CIFTI (and possibly a bunch of other formats) in order to determine the sampling rate.

Reply:
+1 for the structure, here, and having `SamplingFrequency` in the json for all file formats !

Reply:
I think required for all formats is good for consistency. Note that we are likely to have sidecars like `task-<label>_timeseries.json` that will apply equally well to TSV, NIfTI, GIFTI and CIFTI-2 time series. e.g., fMRIPrep has no derivative time series that aren't 1-per-TR, so I expect we'll have a boilerplate `{"SamplingFrequency": "TR"}` for each task.